File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    one = 0
    two = 0
    result = []

    for i in range(len(goal)):
        if(one < len(cards1) and goal[i] == cards1[one]):
            result.append(goal[i])
            one = one + 1
        elif(two < len(cards2) and goal[i] == cards2[two]):
            result.append(goal[i])
            two = two + 1
        else:
            return "No"

    if(goal == result):
        return "Yes"
    else:
        return "No"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    one = 0;
    two = 0;
    result = [];

    for (i = 0; i < goal.length; i++) {
        if (one < cards1.length && goal[i] === cards1[one]) {
            result.push(goal[i]);
            one++;
        } else if (two < cards2.length && goal[i] === cards2[two]) {
            result.push(goal[i]);
            two++;
        } else {
            return "No";
        }
    }

    if (result.join('') == goal.join('')) {
        return "Yes";
    } else {
        return "No";
    }
}